hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/web/URLConnectionFactory.java

```diff
@@ -120,7 +136,7 @@ public HttpURLConnection configure(HttpURLConnection connection)
       }
     }
 
-    return conn;
+    return new BasicAuthConfigurator(conn, basicAuthCredentials);
```

Comment:
should we have something like:
```
private ConfigurationConfigurator getConfigurator(ConfigurationConfigurator configurator, String basicAuthCred) {
if(basicAuthCred != null && !basicAuthCred.isEmpty()) {
return new BasicAuthConfigurator(configurator, basicAuthCred);}
 else {return configurator;}
}
```
and instead of creating new object of basicAuthConfigurator, we call this new method. And if an object of BasicAuthConfigurator can be made, method will return that new object.
What you say?

Reply:
Sounds reasonable! I've made the change.